packages/angular/src/generators/ngrx/lib/validate-options.ts

```diff
@@ -29,7 +38,11 @@ export function validateOptions(
     const nodes = tsquery(ast, NG_MODULE_DECORATOR_SELECTOR, {
       visitAllChildren: true,
     });
-    if (nodes.length === 0) {
+    if (
+      nodes.length === 0 &&
+      (lt(angularVersion, '14.1.0') ||
+        (installedNgRxVersion && installedNgRxVersion.major < 15))
+    ) {
```

Comment:
It's better to do the version check first, before doing any AST parsing. It's a cheaper operation and when it's `false`, we save ourselves the AST parsing. That's why the previous code was first checking the versions and then checking the AST only if needed.